manual/manual/tutorials/coreexamples.etex

```diff
@@ -677,6 +677,68 @@ print_expr e; print_newline ();;
 print_expr (deriv e "x"); print_newline ();;
 \end{caml_example}
 
+\section{Printf formats}
+
+There is a "printf" function in
+\href{https://caml.inria.fr/pub/docs/manual-ocaml/libref/Printf.html}{Printf}
+module (see chapter~\ref{c:moduleexamples}) that allows you to make formatted
+output more concisely.
+It follows the behavior of the "printf" function from the C standard library.
+The "printf" function takes a format string that describes the output values and
+their format, and arguments for output that correspond to the formatting
+directives in the format string.
+\begin{caml_example}{toplevel}
+Printf.printf "%i + %i is an integer value, %F * %F is a float, %S\n"
+3 2 4.5 1. "this is a string";;
+\end{caml_example}
+OCaml type system checks that the type of the arguments and the specifiers are
+compatible. If you pass it an argument of a type that does not correspond to
+the format specifier, the compiler will display an error message:
+\begin{caml_example}{toplevel}[error]
+Printf.printf "Float value: %F" 42;;
+\end{caml_example}
+Associated with this feature is the need to explicitly indicate that the
+string is a format string in order to pass it to the "printf" function.
+\begin{caml_example*}{toplevel}
+let str : _ format =
+    "%i is an integer value, %F is a float, %S\n";;
+\end{caml_example*}
+\begin{caml_example}{toplevel}
+Printf.printf str 3 4.5 "string value";;
+\end{caml_example}
+"fprintf" function is like "printf" except that it takes an output channel as
+the first argument. Here is how to rewrite the pretty-printer using "fprintf":
```

Comment:
Before the big example, it would be nice to have an easy example of fprintf in action, and then an easy example of `%a` in action. Here would be possible suggestions:

```
let output_int ppf n = Printf.fprintf ppf "%d" n

let output_option printer ppf = function
  | None -> Printf.fprintf ppf "None"
  | Some v -> Printf.fprintf ppf "Some(%a)" printer v

let () = Printf.fprintf stdout
  "The current setting is %a"
  (output_option output_int) (Some 3)
```